pandas/io/sas/sas.pyx

```diff
@@ -426,8 +426,10 @@ cdef class Parser:
                 jb += 1
             elif column_types[j] == column_type_string:
                 # string
-                string_chunk[js, current_row] = np.array(source[start:(
```

Comment:
is the perf bottleneck in the `np.array` call?

Reply:
looks like both the .tobytes and the .rstrip are non-optimized.  could cython optimize it if the bytes object were declared as such?

Reply:
Let me check. IIRC tobytes will never be optimized and it’s very slow. The call to np.array is completely useless so I suggest to remove that in any case.

Reply:
Here are some benchmarks on 2 files, times in ms for min duration of 10 reads of each file:


- Baseline: 7.6 / 15
- Remove redundant `np.array()`: 6.6 / 12.7 (0.87x / 0.85x)
- Additionally decare the slice as `cdef bytes`: 6.5 / 12.5
- Additionally use `rfind` plus manually slicing instead of `rstrip`: 6.4 / 11.5
- My solution: 6.2 / 10.6 (0.82x / 0.71x)

Reply:
Will do!